core/modules/layout/layout.flexible.inc

```diff
@@ -488,24 +488,71 @@ function layout_flexible_template_edit_row_form($form, &$form_state, LayoutFlexi
 
   $form['region_names'] = array(
     '#type' => 'fieldset',
-    '#title' => t('Region names'),
+    '#title' => t('Regions'),
     '#collapsed' => $original_row != 'add',
     '#collapsible' => TRUE,
     '#tree' => TRUE,
   );
 
   $form_state['last_region_number'] = layout_flexible_template_get_last_region_number($flexible_template);
   for ($i = 0; $i < $region_count; $i++) {
-    $form['region_names']['region_name_' . $i] = array(
+    $form['region_names']['region_' . $i] = array(
+      '#type' => 'container',
+    );
+    if (isset($row_data['region_names']['region_' . $i]['region_name'])) {
+      $region_name = $row_data['region_names']['region_' . $i]['region_name'];
+    }
+    $form['region_names']['region_' . $i]['region_name'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Region name') . ' ' . ($i + 1),
+      '#default_value' => !empty($region_name) ? $region_name : t('Region') . ' ' . ($form_state['last_region_number'] + $i + 1),
+    );
+    $form['region_names']['region_' . $i]['class_name'] = array(
```

Comment:
Rather than `region_name` and `class_name`, could we use "name" as the machine name, which lines up with things like Views and Layouts themselves? "name" _usually_ means machine name. While we typically use "Title" or "Label" for the human-readable string. So I would suggest the following:

```suggestion
    $form['region_names']['region_' . $i]['label'] = array(
      '#type' => 'textfield',
      '#title' => t('Region label') . ' ' . ($i + 1),
      '#default_value' => !empty($region_name) ? $region_name : t('Region') . ' ' . ($form_state['last_region_number'] + $i + 1),
    );
    $form['region_names']['region_' . $i]['name'] = array(
```